### PR TITLE
Mtdsa 14802 - retire v2

### PR DESCRIPTION
**Pull request by basotia**
https://jira.tools.tax.service.gov.uk/browse/MTDSA-14802

### CODE REVIEW

File: app/api/models/errors/CommonMtdErrors.scala
Comment (GauravJai) [inline app/api/models/errors/CommonMtdErrors.scala:1]: ServiceUnavailableError can be deleted as its unused.

File: test/routing/VersionRoutingRequestHandlerSpec.scala [hunk 1]
```diff
@@ -113,15 +103,15 @@ class VersionRoutingRequestHandlerSpec extends UnitSpec with Inside with MockApp
     "if the request ends with a trailing slash" when {
       "handler found" should {
         "use it" in new Test {
-          MockAppConfig.endpointsEnabled(Version2).returns(true).anyNumberOfTimes()
+          MockAppConfig.endpointsEnabled(Version3).returns(true).anyNumberOfTimes()
```
Comment (darshanbala-hmrc) [inline test/routing/VersionRoutingRequestHandlerSpec.scala:106]: Don't need this line

File: test/routing/VersionRoutingRequestHandlerSpec.scala [hunk 2]
```diff
@@ -113,15 +103,15 @@ class VersionRoutingRequestHandlerSpec extends UnitSpec with Inside with MockApp
...
 
           requestHandler.routeRequest(buildRequest("/")) shouldBe Some(DefaultHandler)
         }
       }
 
       "handler not found" should {
         "try without the trailing slash" in new Test {
-          MockAppConfig.endpointsEnabled(Version2).returns(true).anyNumberOfTimes()
+          MockAppConfig.endpointsEnabled(Version3).returns(true).anyNumberOfTimes()
```
Comment (darshanbala-hmrc) [inline test/routing/VersionRoutingRequestHandlerSpec.scala:114]: Can be removed too

**Review by darshanbala-hmrc — COMMENTED** [review]
Can remove `BackendStub.scala` and we now have `DownstreamStub.scala` in `v4`. Must also ensure all places are using `DownstreamStub` instead

File: app/routing/Versions.scala [hunk 3]
```diff
@@ -55,15 +53,10 @@ sealed trait Version {
...
-case object Version2 extends Version {
-  val name       = "2.0"
-  val configName = "2"
-}
-
 case object Version3 extends Version {
   val name                                    = "3.0"
   val configName                              = "3"
-  override val maybePrevious: Option[Version] = Some(Version2)
+  override val maybePrevious: Option[Version] = None
```
[the remark this answers is not in the document]
Comment (darshanbala-hmrc) [reply app/routing/Versions.scala:59]: Yep, but that particular line is falling back onto nothing - so can be removed. Can be left as is for Version4 case object

File: it/v4/endpoints/RetrieveCalculationControllerISpec.scala [hunk 4]
```diff
@@ -23,8 +23,8 @@ import play.api.http.Status._
 import play.api.libs.json.{JsValue, Json}
 import play.api.libs.ws.{WSRequest, WSResponse}
 import play.api.test.Helpers.AUTHORIZATION
+import stubs.{AuditStub, AuthStub, BackendStub, MtdIdLookupStub}
```
Comment (darshanbala-hmrc) [inline it/v4/endpoints/RetrieveCalculationControllerISpec.scala:26]: Still using `BackendStub`. Can use `DownstreamStub` instead and deleted the other

File: app/routing/VersionRoutingRequestHandler.scala [hunk 5]
```diff
@@ -85,6 +85,7 @@ class VersionRoutingRequestHandler @Inject() (versionRoutingMap: VersionRoutingM
       .handlerFor(request)
       .orElse {
         if (request.path.endsWith("/")) {
+          println(router.documentation)
```
Comment (darshanbala-hmrc) [inline app/routing/VersionRoutingRequestHandler.scala:88]: I think you forgot to remove this line

**Review by darshanbala-hmrc — APPROVED** [review]

**Pull request merged**